### PR TITLE
Async core and import-once example

### LINKED ISSUE #35: merging in include-once?
Hey, I forked this to create https://github.com/nfriedly/docpad-plugin-nodesassimportonce because I though it would be a fairly involved conversion. However it looks like I could have gotten my actual need met with just this module and a bit of configuration, so, I think I'm going to kill mine and just point to this one.

That said, I thought I'd offer to make a PR with include-once as a boolean config option or alternatively just an example that shows how to wire it up without any changes to this lib. Would you be interested in either of those?

Also, I made everything fully asynchronous in my fork, would you like that change included in the PR as well?

### PR DESCRIPTION
**Pull request by nfriedly**
Makes codebase fully async, and also adds an example to the readme for working with the import-once module.

Note: this is branched off of my earlier PR (#34) because I believe that there would be conflicts otherwise. However, I'm targeting `develop` with this PR.

Fixes #35

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,7 +25,8 @@
     "Jimmy King <hello@jimmyking.me> (https://github.com/jking90)",
     "Jan Kolkmeier <jankolkmeier@gmail.com> (https://github.com/jouz)",
     "Benjamin Lupton <b@lupton.cc> (https://github.com/balupton)",
-    "Johannes Troeger (https://github.com/johannestroeger)"
+    "Johannes Troeger (https://github.com/johannestroeger)",
+    "Nathan Friedly (http://nfriedly.com/)"
```
Comment (jking90) [inline package.json]: These are alphabetical by last name

Comment (nfriedly) [inline package.json]: Oh, right, one sec

**Comment by jking90** [comment]
Published in v2.8.0

**Comment by nfriedly** [comment]
Whoot!

**Pull request merged**